src/hotspot/share/nmt/mallocTracker.hpp

```diff
@@ -55,6 +55,8 @@ class MemoryCounter {
  public:
   MemoryCounter() : _count(0), _size(0), _peak_count(0), _peak_size(0) {}
 
+  inline void set_size_and_count(size_t size, size_t count) { _size = size; _count = count; }
```

Comment:
It occurred to me that this invalidates the peak counter. You need to update that too, e.g.

```
void set_size_and_count(size_t size, size_t count) { 
  _size = size;
  _count = count;
  update_peak(size, count);
}
```